apps/lens_blur/lens_blur_generator.cpp

```diff
@@ -0,0 +1,301 @@
+                   this_pixel_is_within_bokeh_of_sample,
+                   1.0f, 0.0f);
+
+        sample_x = x + sample_locations(x, y, s)[0];
+        sample_y = y + sample_locations(x, y, s)[1];
+        output(x, y, c) += sample_weight(x, y, s) * input_with_alpha(sample_x, sample_y, c);
+
+        // Normalize
+        Func final("final");
+        final(x, y, c) = output(x, y, c) / output(x, y, 3);
+
+        /* THE SCHEDULE */
+        if (auto_schedule) {
+            // Provide estimates on the input image
+            left_im.dim(0).set_bounds_estimate(0, 1536);
+            left_im.dim(1).set_bounds_estimate(0, 2560);
+            left_im.dim(2).set_bounds_estimate(0, 3);
+            right_im.dim(0).set_bounds_estimate(0, 1536);
+            right_im.dim(1).set_bounds_estimate(0, 2560);
+            right_im.dim(2).set_bounds_estimate(0, 3);
+            // Provide estimates on the parameters
+            slices.set_estimate(32);
+            focus_depth.set_estimate(13);
+            blur_radius_scale.set_estimate(0.5f);
+            aperture_samples.set_estimate(32);
+            // Provide estimates on the pipeline output
+            final.estimate(x, 0, 1536)
+                .estimate(y, 0, 2560)
+                .estimate(c, 0, 3);
+            // Auto schedule the pipeline
+            Pipeline p(final);
+            p.auto_schedule(get_target());
+        } else if (get_target().has_gpu_feature()) {
+            Var xi("xi"), yi("yi"), zi("zi");
+            cost_pyramid_push[0].compute_root()
+                .reorder(c, z, x, y)
+                .bound(c, 0, 2)
+                .unroll(c)
+                .gpu_tile(x, y, xi, yi, 16, 16);
+            cost.compute_at(cost_pyramid_push[0], xi);
+            cost_confidence.compute_at(cost_pyramid_push[0], xi);
+
+            for (int i = 1; i < 8; i++) {
+                cost_pyramid_push[i].compute_root()
+                    .gpu_tile(x, y, z, xi, yi, zi, 8, 8, 8);
+                cost_pyramid_pull[i].compute_root()
+                    .gpu_tile(x, y, z, xi, yi, zi, 8, 8, 8);
+            }
+
+            depth.compute_root()
+                .gpu_tile(x, y, xi, yi, 16, 16);
+            input_with_alpha.compute_root()
+                .reorder(c, x, y).unroll(c).gpu_tile(x, y, xi, yi, 16, 16);
+            worst_case_bokeh_radius_y
+                .compute_root()
+                .gpu_tile(x, y, xi, yi, 16, 16);
+            worst_case_bokeh_radius
+                .compute_root()
+                .gpu_tile(x, y, xi, yi, 16, 16);
+            final.compute_root()
+                .reorder(c, x, y)
+                .bound(c, 0, 3)
+                .unroll(c)
+                .gpu_tile(x, y, xi, yi, 16, 16);
+
+            output.compute_at(final, xi);
+            output.update().reorder(c, x, s).unroll(c);
+            sample_weight.compute_at(output, x);
+            sample_locations.compute_at(output, x);
+        } else {
```

Comment:
The comments in the section don't make sense now that they've been lifted out of the log I kept while scheduling. Better to just replace them with // Manual CPU schedule